spec/services/create_requested_account_spec.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe CreateRequestedAccount, focus: true do
+  let(:creator) { create(:admin) }
+  let(:course) { create(:course) }
+  let(:user) { create(:user) }
+  let(:requested_account) { create(:requested_account, course_id: course.id, username: user.username, email: user.email) }
+  let(:subject) do
+    described_class.new(requested_account, creator)
+  end
+
+  it 'creates the requested accounts' do
+    stub_account_creation
+    expect(subject.result[:failure]).to_not be_nil
```

Comment:
@ragesoss here I am getting an error I don't understand.
<img width="953" alt="captura de pantalla 2018-01-23 a las 18 10 53" src="https://user-images.githubusercontent.com/8235980/35290259-f601083e-0069-11e8-9df6-6cbe7bbf80d2.png">

Apparently is raising the error of CreateRequestedAccount line 55 'raise AccountCreationError, "could not create user #{returned_username}" if @user.blank?' but I don't really get if if when we are stubbing the createaccount action we are sending both the createaccount + username action (why is @user blank?). 

It could also be that I am missing something in between. I would appreciate some hints... :) Thanks!

Reply:
@mauditecandela it's because of what happens on line 54:

`@user = UserImporter.new_from_username(returned_username, @wiki)`

The account has been created, but we don't have all the data and username normalization that we normally do for new User records. So instead of creating a User record directly in our database, after creating the Wikimedia acount, we use `UserImporter` to fetch that user's info from Wikipedia (the same way we do with any other new User record).

This is the same problem we solved in the controller spec with `allow(UserImporter).to receive(:new_from_username).and_return(user)`.